llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -4282,6 +4282,38 @@ SelectionDAG::computeOverflowForSignedMul(SDValue N0, SDValue N1) const {
   return OFK_Sometime;
 }
 
+SDNodeFlags SelectionDAG::computeShiftFlags(SDValue Op) const {
+  assert((Op.getOpcode() == ISD::SRL ||  Op.getOpcode() == ISD::SRA ||
+          Op.getOpcode() == ISD::SHL) &&
+         "Expecting an SHL/SRL/SRA operation.");
+  SDValue Op0 = Op.getOperand(0);
+  SDNodeFlags Flags = Op->getFlags();
+
```

Comment:
Skip doing anything if the flags are already set?

Reply:
I've tried to guard the most expensive computations below by checking if the flags already are present. So not quite sure what you propose.